### PR TITLE
feat: Implemented External OAuth Security Integration Resource

### PR DESCRIPTION
**Pull request by ChrisIsidora**
Implemented External OAuth Security Integration Resource thereby solves #874 

## Test Plan
* [x] acceptance tests
* [x] tests

## References

* https://docs.snowflake.com/en/user-guide/oauth-ext-overview.html
* https://docs.snowflake.com/en/sql-reference/sql/create-security-integration.html

### CODE REVIEW

**Comment by ChrisIsidora** [comment]
@alldoami  Can you review?

**Comment by ChrisIsidora** [comment]
I expect that we need this feature the coming week can this be also merged and a new version released? @alldoami / @edulop91.
Thanks in advance

File: examples/resources/snowflake_external_oauth_integration/resource.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+resource "snowflake_external_oauth_integration" "azure" {
+  name                             = "AZURE_POWERBI"
+  type		                       = "AZURE"
+  enabled                          = true
+  issuer					       = "https://sts.windows.net/00000000-0000-0000-0000-000000000000"
+  snowflake_user_mapping_attribute = "LOGIN_NAME"
+  jws_keys_urls					   = ["https://login.windows.net/common/discovery/keys"]
+  audience_urls	 				   = ["https://analysis.windows.net/powerbi/connector/Snowflake"]
+  token_user_mapping_claims        = ["upn"]
+}
```
Comment (alldoami) [inline examples/resources/snowflake_external_oauth_integration/resource.tf:10]: Could you reformat this file?

File: pkg/resources/external_oauth_integration_acceptance_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		name                             = "%s"
+		type		                     = "%s"
+		enabled                          = true
+  		issuer					     	 = "https://sts.windows.net/00000000-0000-0000-0000-000000000000"
+  		snowflake_user_mapping_attribute = "LOGIN_NAME"
+		jws_keys_urls					 = ["https://login.windows.net/common/discovery/keys"]
+		audience_urls	 				 = ["https://analysis.windows.net/powerbi/connector/Snowflake"]
+  		token_user_mapping_claims        = ["upn"]
+	}
+	`, name, integrationType)
```
Comment (alldoami) [inline pkg/resources/external_oauth_integration_acceptance_test.go:52]: Reformat

**Comment by ChrisIsidora** [comment]
@alldoami Fixed Formats and Unit Test

**Review by alldoami — APPROVED** [review]

**Pull request merged**